### PR TITLE
feature/505+483: create recorder popup

### PR DESCRIPTION
**Pull request by csanyiarpad**
- create a block for logged in users to show a record button on all pages except the recorder form
- create a 2-step popup to show when the button is clicked (template, css, js)
  1) enter URL the record
  2) enter information for saving
- rename the label and options of severity to form a sentence
- error messages for filling the form + warning messages about being in alpha... (feature/483)
- saving will redirect to the created node, as the original form
- deleting, and starting a new will reload the page and automatically show the popup prefilled with the default URL

### CODE REVIEW

File: walkhub.features.field_base.inc [hunk 1]
```diff
@@ -438,9 +438,9 @@ function walkhub_field_default_field_bases() {
     'module' => 'list',
     'settings' => array(
       'allowed_values' => array(
-        0 => 'No changes (tour)',
-        1 => 'Changes content (run only on sites where this is ok)',
-        2 => 'Changes configuration (run only on staging or test environments)',
+        0 => 'does not change anything (tour)',
+        1 => 'changes content (run only on sites where this is ok)',
+        2 => 'changes configuration (run only on staging or test environments)',
```
Comment (Coornail) [inline walkhub.features.field_base.inc:443]: I think we changed that intentionally, no?

File: walkhub.module [hunk 2]
```diff
@@ -21,6 +21,39 @@ function walkhub_init() {
...
+  return $blocks;
+}
+
+/**
+ * Implements hook_block_view().
+ */
+function walkhub_block_view($delta = '') {
+  $block = array();
+  $user_roles = $GLOBALS[user]->roles;
+  if (!in_array("authenticated user", $user_roles)) {
```
Comment (tamasd) [inline walkhub.module]: I think using `user_is_anonymous()` would be better here.

File: walkhub.features.field_base.inc (hunk 1 above)
Comment (csanyiarpad) [inline walkhub.features.field_base.inc:443]: I changed it to from a sentence (talked with Dia, Kristof)
This walkthrough: (does not change anything | [option2] | [option3])

File: walkhub.module [hunk 3]
```diff
@@ -1130,6 +1163,11 @@ function walkhub_theme() {
       'base hook' => 'page',
       'path' => drupal_get_path('module', 'walkhub'),
     ),
+    'walkthrough__record__block' => array(
+      'variables' => array("form" => drupal_get_form('walkhub_record_form', true)),
```
Comment (Coornail) [inline walkhub.module]: `TRUE` should be big capital.

File: walkthrough_record_block.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  $(document).ready(function() {
+
+    var url_re = /^http\:\/\/[a-zA-Z0-9\-\.]+\.[a-zA-Z]+(\/\S*)?$/;
+
+    $("#record-a-walkthrough-button").click(function (e) {
+      var url_field = $("#walkthrough-record-url");
+      $(this).hide();
+      $("#record-a-walkthrough-popup-modal").show();
+      if (url_field.val() == "") {
+        url_field.val("http://pronovix.com/contact-us");
```
Comment (Coornail) [inline walkthrough_record_block.js]: Lets have this default value in a constant.

File: walkthrough_record_block.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
+(function($) {
+  $(document).ready(function() {
+
+    var url_re = /^http\:\/\/[a-zA-Z0-9\-\.]+\.[a-zA-Z]+(\/\S*)?$/;
```
Comment (Coornail) [inline walkthrough_record_block.js]: https?

File: walkthrough_record_block.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    $("#walkthrough-start-recording").click(function (e) {
+      var url = $("#walkthrough-record-url").val();
+      if (!url_re.test(url)) {
+        $("#url-error").show();
+      } else {
+        $("#url-error").hide();
+        $("#edit-url").val(url);
+        $("#edit-record").click();
+
+        var title = $("#walkthrough-record-title-wrapper");
```
Comment (Coornail) [inline walkthrough_record_block.js]: We usually prefix jQuery variables with `$`.

File: walkthrough_record_block.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+      if (e.target.innerHTML.length > 0) {
+        stepsBox.val(steps.text().replace(/\)/g,")\n"));
+      }
+    });
+
+    $("#walkthrough-record-save").click(function() {
+      var input_title = $("#walkthrough-record-title");
+      var select_severity = $("#walkthrough-record-severity");
+      var error = false;
+      if (input_title.val() == "") {
```
Comment (Coornail) [inline walkthrough_record_block.js]: #

File: walkthrough_record_block.js [hunk 8]
```diff
@@ -0,0 +1,109 @@
+(function($) {
```
Comment (Coornail) [inline walkthrough_record_block.js:1]: Let's make sure this file conforms our global eslint assertions: https://github.com/Pronovix/WalkHub-resources/blob/master/.eslintrc

**Pull request merged**